### PR TITLE
AC-392:Download patient option not switching instantly

### PR DESCRIPTION
**Pull request by DefCon-007**
JIRA issue link : https://issues.openmrs.org/browse/AC-392

### CODE REVIEW

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/ACBaseActivity.java [hunk 1]
```diff
@@ -49,7 +49,7 @@
     protected final OpenMRSLogger mOpenMRSLogger = mOpenMRS.getOpenMRSLogger();
     protected AuthorizationManager mAuthorizationManager;
     protected CustomFragmentDialog mCustomFragmentDialog;
-    private MenuItem mSyncbutton;
+    private MenuItem mSyncbutton,mAddPatientMenuItem;
```
Comment (tmarzeion) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/ACBaseActivity.java]: I think we should use new line for each field (I think it increases readability)

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/ACBaseActivity.java [hunk 2]
```diff
@@ -83,12 +83,16 @@ public boolean onCreateOptionsMenu(Menu menu) {
     @Override
     public boolean onPrepareOptionsMenu(final Menu menu) {
         mSyncbutton = menu.findItem(R.id.syncbutton);
+        mAddPatientMenuItem = menu.findItem(R.id.actionAddPatients);
```
Comment (tmarzeion) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/ACBaseActivity.java]: Its really bad behavior to add `mAddPatientMenuItem` in this class.
`ACBaseActivity` is not intended to handle behavior like this and it should be solved by using some listener between `mAddPatientMenuItem` to `mSyncbutton` inside `SyncedPatientsFragment` - not `ACBaseActivity`.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/ACBaseActivity.java [hunk 3]
```diff
@@ -138,6 +145,12 @@ public boolean onOptionsItemSelected(MenuItem item) {
         }
     }
 
+    private void enableAddPatient(boolean enabled) {
+        int resId = enabled ? R.drawable.ic_add : R.drawable.ic_add_disabled;
+        mAddPatientMenuItem.setEnabled(enabled);
```
Comment (tmarzeion) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/ACBaseActivity.java]: Gives `NPE` when using sync button in main dashboard (or anywhere else where `mAddPatientMenuItem` isn't defined)
Dont check if `mAddPatientMenuItem` is null to workaround this. Get rid of `mAddPatientMenuItem` from `ACBaseActivity` instead.

**Comment by DefCon-007** [comment]
@tmarzeion I tried to use the `onOptionsItemSelected` in `SyncedPatientsFragment` but for some reasons when I click the sync button it is not registered by the `SyncedPatientsFragment`

**Comment by tmarzeion** [comment]
@DefCon-007 
I've moved menu logic to Activity. It may be a bit hacky when using MVP, but I think its easiest solution :)
Do it similar to https://github.com/openmrs/openmrs-contrib-android-client/pull/365 :)

**Comment by DefCon-007** [comment]
@AvijitGhosh82 @tmarzeion For some reasons now in the `SyncedPatientsActivity` back button is not working.

**Pull request merged**